### DISCUSSION TITLE
Video on czech television

### DISCUSSION THREAD

**Discussion by dubat3223** [aclap-dev/video-downloadhelper#610] [Help and Q&A]
### Your question

There was an error in downloading the identified video on ceskatelevize.cz 

Here are the copied bug details: 
{
  "version": "9.0.1.13",
  "target": "mozilla",
  "channel": "stable",
  "page_url": "https://www.ceskatelevize.cz/porady/15159159462-zapomenute-pribehy-2-svetove-valky/223382576500006/",
  "error": {
    "error": "coapp_failure",
    "details": "SideDownload error: : [in#0 @ 00000202ba2eb200] Error opening input: Server returned 404 Not Found\r\nError opening input file https://ivys-vst23-5.o2tv.cz/atip/e8782e0de83bbc49916911410e0065ae/1716562446897/61924494878222986-pc.mpd?userId=931db6de-1ace-40de-acf7-d53acce9cc15.\r\nError opening input files: Server returned 404 Not Found\r\n",
    "id": "coapp_failure_e7e69887-0cbf-4f41-a2ac-80d8b25c0a4e",
    "downloadable_id": "downloadable_1680124788221182"
  }
}

### Addon & CoApp version and details

Video DownloadHelper
version 9.0.1.13
updated 23rd May, 2024

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I am just another user.

I am running Windows 11 Pro 64-bit version 23H2 build 22631.3593, Firefox 126.0 64-bit, licensed VDH 9.0.1.22a1 beta, CoApp 2.0.19.

I visited the web page whose URL you provided in page_url above.  VDH recognized nothing until I launched playback in the player on the web page.  As soon as I did that, VDH did recognize something as MPD.  That indicates that the transport mechanism here is DASH.  At the same time, the icon I'm highlighting in the following image appeared.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/bb3ce142-6a48-4596-96b1-87d7b44ed351)

This content is protected by DRM.  That means that VDH will never succeed in downloading this content.  I tried it & I got the same error you did.

BUT

I decided to go to Plan B anyway.  I thought, what could it hurt?  You can see in that image that the Network Monitor offered a DASH master manifest.  MPD files are DASH manifests.  The one here happened to be a master manifest.  A master manifest contains information for finding stream manifests.  In this case, there happened to be information for 5 video-only streams of various resolutions & 1 audio-only stream.  I then executed ffmpeg to download the 1920x1080 video-only stream along with the audio-only stream.  It was downloading at a putrid speed, slow enough that it would have taken over 2 hours to download this video that is less than 1 hour long.  So I interrupted the download.  I was astonished to discover that the few seconds I had downloaded with ffmpeg played just fine in VLC, both video & audio.

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/0d08876e-ca49-4ce9-92b0-ad10d6ddf4ce)

This is entirely unexpected.  When content is protected by DRM, what both VDH & ffmpeg download is supposed to be scrambled.  The image is supposed to be unwatchable & there is supposed to be no audio at all.  The only explanation I can give is that this web site has incorrectly implemented DRM.

Now I find it very unexpected that VDH was unable to download this content but I was able to use ffmpeg to download it no problem.  Further, the ffmpeg I use is the one that is distributed with the CoApp.  It's the same one VDH uses when it downloads content.  So I can't explain this failure by VDH when VDH's own ffmpeg had no problem with it.

If you want to download this video, & I would imagine others like it, you will learn how to use ffmepg.  I posted a tutorial on ffmpeg in our old Google forum.  Do a Google search using "vdh ffmpeg" as your search key.  The first result is a link to my tutorial.

Happy reading!

**Reply by GrampaWildWilly** [reply]
Note to @paulrouget Paul.  The DASH master manifest actually does not have MPD in its file name.  But I do all my ffprobes & ffmpegs with -allowed_extensions ALL.  Maybe I'm avoiding an error VDH is hitting.  Just a guess.
